Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FitGaussianPeaks.py

```diff
@@ -303,10 +320,16 @@ def general_fit(self, xvals, yvals, peakids):
         fit_func = ''
         fit_constr = ''
         for i, pid in enumerate(peakids):
-            win_size = int(self._estimate_sigma * len(xvals) / (max(xvals) - min(xvals)))
-            if win_size < 2:
-                win_size = 2
-            centre, height, sigma = tuple(self.estimate_single_parameters(xvals, yvals, pid, win_size))
+            if self._estimate_fit_window:
+                win_size = int(self._estimate_sigma * len(xvals) / (max(xvals) - min(xvals)))
+                if win_size < 2:
+                    win_size = 2
+            else:
+                win_size = int((self._fit_window - 1)/2)
```

Comment:
Could we extract this into its own method, since `general_fit` is gradually growing:

`win_size = _get_window_size()`